### PR TITLE
Feature/kas 4322 pub search for all

### PR DESCRIPTION
**Pull request by sergiofenoll**
https://kanselarij.atlassian.net/browse/KAS-4322

**Overview**
Adds a new search route for all profiles to search based for publication flows.

This new route supports all global search filters (date & mandatees) as well as filtering on the publication status.
Sorting happens on the vergadering geplande startdatum or on score relevance as determined by Elasticsearch.
This also includes a new result card for publication flows.

**Instructions**
Either create a new agenda or use an existing agenda. Add an agenda item **that is linked to a subcase** to this agenda and upload a piece. Create a publication flow for this piece using the "Documenten voor publicatie" button. Finalise the agenda and close the meeting. Now the agenda should propagate to the other profiles, and depending on the access level of the piece some profiles will already see it, but none will see the publication flow. Now **release the decisions**, after doing this Yggdrasil should pick up the changes and propagate the publication flow. Make sure to also release the documents if you're checking for a different profile than "Minister" (i.e. Overheidsorganisatie will only see documents after the document release, whereas Minister will always see documents).

**Related PRs:**
- [ ] https://github.com/kanselarij-vlaanderen/app-kaleidos/pull/427
- [ ] https://github.com/kanselarij-vlaanderen/yggdrasil/pull/59

### CODE REVIEW

File: app/components/search/publication-statuses-filter.hbs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<Auk::Label>{{t "publication-status"}}</Auk::Label>
+{{#if this.loadPublicationStatuses.isRunning}}
+  <Auk::Loader />
+{{else}}
+  <Auk::CheckboxTree
+    @label={{capitalize (t "all-results")}}
```
Comment (kobemertens) [inline app/components/search/publication-statuses-filter.hbs]: Maybe change to "Alle Statussen"?

File: app/components/search/publication-statuses-filter.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  loadPublicationStatuses = task(async () => {
+    this.statuses = (await this.store.queryAll(
+      'publication-status',
+      { sort: 'position' },
+    )).toArray();
+
+    if (this.selectedStatusIds.length) {
+      this.selectedStatuses = this.statuses.filter(decisionResultCode =>
```
Comment (kobemertens) [inline app/components/search/publication-statuses-filter.js]: **nitpick**: probably forgot to rename `decisionResultCode`. Not really important

File: app/components/search/result-cards/publication-flow.hbs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      {{#if @decisionmakingFlowId}}
+        <AuLink
+          class="au-u-padding-none card-link"
+          @route="cases.case.subcases"
+          @model={{@decisionmakingFlowId}}
+        >
+          <span class="card-link">
+            <SanitizeHtml
+              @raw={{true}}
+              @value={{if @shortTitle @shortTitle @title}}
```
Comment (kobemertens) [inline app/components/search/result-cards/publication-flow.hbs]: **nitpick**: Could use `or @shortTitle @title` here, not sure if more readable

File: app/components/search/result-cards/publication-flow.hbs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            />
+          </span>
+        </AuLink>
+      {{else}}
+        <SanitizeHtml
+          @raw={{true}}
+          @value={{if @shortTitle @shortTitle @title}}
```
Comment (kobemertens) [inline app/components/search/result-cards/publication-flow.hbs]: Same here

**Review by ValenberghsSven — APPROVED** [review]
tested / changed where needed in combination with government areas.
Added plausible for this new search route.
Fixed status pills not showing correct icons.

**Pull request merged**